frontend/src/stores/ethereum-provider.ts

```diff
@@ -0,0 +1,19 @@
+import { JsonRpcSigner } from '@ethersproject/providers';
+import { defineStore } from 'pinia';
+import { shallowRef } from 'vue';
+
+import { EthereumProvider } from '@/services/web3-provider';
+
+export const useEthereumProvider = defineStore('ethereumProvider', {
+  state: () => ({
+    // We need a shallow reference here to prevent issues with ethers.js on
+    // contract function calls.
+    provider: shallowRef<EthereumProvider | undefined>(undefined),
+  }),
+  getters: {
+    isConnected: (state) => state.provider !== undefined,
```

Comment:
We had a good meeting together to get this sorted out. The result was to go back to the old `signer` and `provider` checks which can be undefined or be defined. The `isConnected` got mixed up and confused. The problem here is that with the future upcoming multiple ethereum provider options this will change a lot. We will re-evaluate this again once we reached this state.
I pushed the related fixup commit.